cmd/dao_tests/resource_test.go

```diff
@@ -0,0 +1,165 @@
+//go:build integration
+// +build integration
+
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	_ "github.com/RHEnVision/provisioning-backend/internal/dao/sqlx"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func createPkResource() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeNoop,
+		Handle:   "handle1",
+	}
+}
+
+func createPkResource2() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeAzure,
+		Handle:   "handle1",
+	}
+}
+
+func setupResource(t *testing.T) (dao.PubkeyResourceDao, context.Context) {
+	setup()
+	ctx := identity.WithTenant(t, context.Background())
+	resourceDao, err := dao.GetPubkeyResourceDao(ctx)
+	if err != nil {
+		panic(err)
+	}
+	return resourceDao, ctx
+}
+
+func teardownPubkeyResource(_ *testing.T) {
+	teardown()
+}
+
+func TestCreateResource(t *testing.T) {
+	resourceDao, ctx := setupResource(t)
+	defer teardownPubkeyResource(t)
+	resource := createPkResource()
+	err := resourceDao.Create(ctx, resource)
+	if err != nil {
+		panic(err)
+	}
+	resources, err := resourceDao.ListByPubkeyId(ctx, resource.PubkeyID)
+	if err != nil {
+		panic(err)
+	}
+
+	assert.Equal(t, 1, len(resources))
+	assert.Equal(t, resource.ID, resources[0].ID)
+	assert.Equal(t, resource.Provider, resources[0].Provider)
+	assert.Equal(t, resource.Handle, resources[0].Handle)
+	assert.Equal(t, resource.Tag, resources[0].Tag)
+}
+
+func TestGetResourceByProviderType(t *testing.T) {
+	resourceDao, ctx := setupResource(t)
+	defer teardownPubkeyResource(t)
+	resource := createPkResource()
+	err := resourceDao.Create(ctx, resource)
+	if err != nil {
+		panic(err)
+	}
+	createdResource, err := resourceDao.GetResourceByProviderType(ctx, resource.PubkeyID, resource.Provider)
+	if err != nil {
+		panic(err)
+	}
+
+	assert.Equal(t, resource.ID, createdResource.ID)
+	assert.Equal(t, resource.Provider, createdResource.Provider)
+	assert.Equal(t, resource.Handle, createdResource.Handle)
+	assert.Equal(t, resource.Tag, createdResource.Tag)
+}
+
+func TestListByPubkeyIdResource(t *testing.T) {
+	resourceDao, ctx := setupResource(t)
+	defer teardownPubkeyResource(t)
+	pkId := int64(1)
+	resourcesBefore, err := resourceDao.ListByPubkeyId(ctx, pkId)
+	if err != nil {
+		panic(err)
+	}
+	err = resourceDao.Create(ctx, createPkResource())
+	if err != nil {
+		panic(err)
+	}
+	err = resourceDao.Create(ctx, createPkResource2())
+	if err != nil {
+		panic(err)
+	}
+	resourcesAfter, err := resourceDao.ListByPubkeyId(ctx, pkId)
+	if err != nil {
+		panic(err)
+	}
+
+	assert.Equal(t, len(resourcesBefore)+2, len(resourcesAfter))
```

Comment:
I would suggest to also assert that the returned list actually contains both resources. You can take advantage of `require.Contains` which works with slices I think.